### DISCUSSION TITLE
I need to deploy the dataExtensions which i was retrieved from one Business Unit to Other Business Unit

### DISCUSSION THREAD

**Discussion by pothangelli123** [Accenture/sfmc-devtools#1223] [Q&A] [answered]
Subject: The retrieved data in which of any form(Data Extension, Asset , etc) from a business unit of a installed package is not able to deploy to other Business unit where there are no external security reasons are involved so that moving of data from one BU to other BU is not being performed.
Commands used: 
mcdev init
// selected a business unit under some credentials of sfmc instance.
mcdev selectTypes 
asset or data extension
mcdev r // for retrieving data from the business unit 
//after succesfully retrieving data from the business unit, cant able to move it or deploy it to other business.as per organizational norms i cant able to specify the names of business units but i need help in doing so.
mcdev d <BU> <TYPE> <KEY>
ERROR STATUS im getting in my terminal:
You could directly pass in this info with 'pothanpackage/DEPLOY1'
02:25:11 error: mcdev.deploy failed:
02:25:11 error:     No metadata found in deploy folder for selected BU & type (undefined)
02:25:11 info: :: Deployed 1 BUs
 other commands i used are: 
 mcdev d 
 after selecting the business unit for as the destination:
 You could directly pass in this info with 'pothanpackage/DEPLOY1'
02:19:56 info: :: Deploying to pothanpackage/DEPLOY1
02:19:56 info: Caching dependent Metadata: folder
02:19:56 info:  - Subtypes: hidden, dataextension, salesforcedataextension, shared_data, shared_dataextension, shared_salesforcedataextension, synchronizeddataextension, mysubs, list, publication
02:20:00 info: Caching dependent Metadata: dataExtensionTemplate
02:20:01 info: Caching dependent Metadata: dataExtension
02:20:02 info: Caching dependent Metadata: attributeSet
02:20:06 info: Caching dependent Metadata: attributeGroup
02:20:06 info: Caching dependent Metadata: fileLocation
02:20:07 info: Caching dependent Metadata: list
02:20:08 info:  - Caching dependent Metadata: folder
02:20:08 info:  - Subtypes: list, mysubs, suppression_list, publication, contextual_suppression_list
02:20:09 info: Deploying: dataExtension
02:20:09 info:  - Caching dependent Metadata: dataExtensionField
02:20:11 error:  ☇ skipping dataExtension 0ED8D7CD-972A-4CA2-83CD-1804F58ACF51 / _MobileAddressApplication: Cannot Upsert Strongly Typed Data Extensions
02:20:11 info: dataExtension upsert: 0 of 0 created / 0 of 0 updated / 1 filtered
02:20:11 info: Deploying: attributeSet
02:20:11 error:  ☇ skipping attributeSet ChatMessageDemographics / Chat Message Demographics: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet ChatMessageSubscriptions / Chat Message Subscriptions: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet Contact / Contact: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet EinsteinMCPredictiveScores / Einstein MC Predictive Scores: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet EmailAddresses / Email Addresses: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet EmailDemographics / Email Demographics: update is not supported yet for attributeSet
Time: 23.043s
02:20:11 error:  ☇ skipping attributeSet GroupConnectLineAddress / GroupConnect LINE Addresses: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet GroupConnectLineDemographics / GroupConnect LINE Demographics: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet GroupConnectLineSubscriptions / GroupConnect LINE Subscriptions: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet MobileDemographics / MobileConnect Demographics: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet MobileSubscriptions / MobileConnect Subscriptions: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelAbandonedCartEvents / Predictive Intelligence AbandonedCartEvents: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelAbandonedCartItems / Predictive Intelligence AbandonedCartItems: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelContent / Predictive Intelligence Content: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelContentAttribs / Predictive Intelligence Content Attributes: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelContentViews / Predictive Intelligence Content Views: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelProductAttribs / Predictive Intelligence Product Attributes: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelProductPurchases / Predictive Intelligence Product Purchases: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelProductSessions / Predictive Intelligence Product Sessions: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelProductViews / Predictive Intelligence Product Views: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelProducts / Predictive Intelligence Products: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelProfiles / Predictive Intelligence Profiles: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PredictiveIntelSessionEnds / Predictive Intelligence Session Ends: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PushDemographics / MobilePush Demographics: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PushSubscriptions / MobilePush Subscriptions: update is not supported yet for attributeSet
02:20:11 error:  ☇ skipping attributeSet PushTags / MobilePush Tags: update is not supported yet for attributeSet
02:20:11 info: attributeSet upsert: 0 of 0 created / 0 of 26 updated
02:20:11 info: Deploying: attributeGroup
02:20:11 error:  ☇ skipping attributeGroup ETChatMessage / undefined: update is not supported yet for attributeGroup
02:20:11 error:  ☇ skipping attributeGroup ETEmail / undefined: update is not supported yet for attributeGroup
02:20:11 error:  ☇ skipping attributeGroup ETGroupConnectLine / undefined: update is not supported yet for attributeGroup
02:20:11 error:  ☇ skipping attributeGroup ETMobileConnect / undefined: update is not supported yet for attributeGroup
02:20:11 error:  ☇ skipping attributeGroup ETMobilePush / undefined: update is not supported yet for attributeGroup
02:20:11 error:  ☇ skipping attributeGroup ETPredictiveIntel / undefined: update is not supported yet for attributeGroup
02:20:11 error:  ☇ skipping attributeGroup ETSystem / undefined: update is not supported yet for attributeGroup
02:20:11 info: attributeGroup upsert: 0 of 0 created / 0 of 7 updated
02:20:11 info: Deploying: fileLocation
02:20:11 error:  ☇ skipping fileLocation ExactTarget Enhanced FTP / ExactTarget Enhanced FTP: update is not supported yet for fileLocation
02:20:11 error:  ☇ skipping fileLocation Salesforce Objects & Reports / Salesforce Objects & Reports: update is not supported yet for fileLocation
02:20:11 info: fileLocation upsert: 0 of 0 created / 0 of 2 updated
02:20:11 info: Deploying: list
02:20:11 error:  ☇ skipping list All Subscribers - 99783 / All Subscribers: update is not supported yet for list
02:20:11 info: list upsert: 0 of 0 created / 0 of 1 updated
02:20:11 info: :: Deployed 1 BUs
  So can you make me any suggestions in this scenario or just make any documentations clearly .
  @oscarrenalias 
  @gdman @JoernBerkefeld @jredding @deors 
please help !!!!!

**Comment by JoernBerkefeld** [comment]
Hm, that's incredibly hard to read but let me try my best step by step:

**part 1**
```
mcdev d
ERROR STATUS im getting in my terminal:
You could directly pass in this info with 'pothanpackage/DEPLOY1'
02:25:11 error: mcdev.deploy failed:
02:25:11 error: No metadata found in deploy folder for selected BU & type (undefined)
02:25:11 info: :: Deployed 1 BUs
```

This basically says you tried to deploy to a BU for which there was nothing in your `/deploy/pothanpackage/DEPLOY1` folder.


**part 2**

```
update is not supported yet for ...
```

Then, it seems you tried to deploy a bunch of things for which deploying is not supported. Attribute Set, attribute group, file location, a special data extension

What exactly did you WANT to deploy in the first place? There is an overview of supported metadata types including details on support for Update/create

**Comment by pothangelli123** [comment]
Thanks for replying sir @JoernBerkefeld , i tried to retrieve an asset type from one business unit and it was successful. But i need to deploy this retrieved asset to other business unit. So i used command "mcdev d", after using this command i need to select the business unit as my destination BU, So that the retrieved asset needed to move to that place. So that im stuck at deploying this asset to other business unit. That's the error I'm getting.

**Reply by JoernBerkefeld** [reply]
i get that. without having to reveal what your solution does, can you please share **what** you are trying to copy/deploy?
examples could be:
- dataExtension
- automation
- fileTransfer
- asset (content builder things)
- ...

full list of types and what is supported for them is listed here: https://github.com/Accenture/sfmc-devtools/wiki/05.-Metadata-Type-Support
If it says "Yes" in the **deploy (create)** or **deploy (update)** column then you are in luck. if it says "-" then you are not.

**Comment by pothangelli123** [comment]
@JoernBerkefeld  sir , the data type of meta data im going to retrieve and deploy are like assets and data extensions. So i just want to specify my entire task , Please dont mind, im new to this tech and trying to learn.
 So my task is to retrieve a data extension first so i used the below command 
 "mcdev init"
 it asked me to enter credentials:
 then i have entered a installed package name in my setup of my sfmc.
 then i given a client id, client server and auth base uri then i have to select the BU.
 Then i ran a command :
 "mcdev selectTypes"
 I have tried for data extension and asset types. i have selected this data type with the help of arrow keys.
 Then in used a command :
 "mcdev r"
 then again it asked for credentials and BU.
 STEP 2 : The next thing is to deploy the retrieved asset from credential 1, BU -1 TO Credential 2 ,BU 2 so I used a command :
 "mcdev d"
  it asked for same thing to choose credential, BU to where the asset is needed to be deloyed.
  SO THIS is the thing im going with or stuck at. And i followed a youtube video shared in our team chat im going to give it here :
  https://youtu.be/Fmp1Dx9yFpw?feature=shared
  So if im doing a wrong process for any of the step feel free to help me and addressing the issue.

**Reply by JoernBerkefeld** [reply]
Hm, not much time this very moment but 2 observations: 

A) you dont need to run selectTypes for standard operations like the ones you described.

B) make sure you copy the folders and json files from your retrieve/credential1/bu/.../ to deploy/credential2/bu/... before running mcdev deploy.

**Reply by JoernBerkefeld** [reply, marked as answer]
ok, a bit more details:

1. you already ran mcdev init... that's a 1-time setup.
2. when you run `mcdev retrieve`, that will download everything that we defined as default types - unless you changed it with `mcdev selectTypes` which you indicated you did. I would suggest you re-run `mcdev selectTypes` and select all types that (some will not be selectable here - ignore those).
3. re-run `mcdev retrieve`, select the cred & bu you want to copy from.
4. open the folder /retrieve/cred1/bu-source/ - note: your folders will have different names. replace "cred1" and "bu-source" with your own values. 
5. identify all types that you want to copy to the other BU and copy their folders (e.g. dataExtension and asset)
6. create a new folder /deploy/cred2/bu-target/ note: your folders will have different names. replace "cred2" and "bu-target" with your own values. 
7. paste the 2 folders you previously copied into this new folder.
8. make sure you check if you just copied data extensions or assets that you did NOT want to copy to the target BU - delete those from the newly created deploy folder!
9. run `mcdev deploy cred2/bu-target` - that should then deploy EVERYTHING that's in /deploy/cred2/bu-target/ folder

**Comment by pothangelli123** [comment]
yes sir thank you, it really worked. But it deployed all the folders from BU to other but the data extensions with definitions are not deploying. Currently i'm working on Specific Automation activities to deploy from one BU to other BU. So i will share the update and thanks for responding.
